sklearn/metrics/_pairwise_distances_reduction/_datasets_pair.pyx.tp

```diff
@@ -158,6 +164,55 @@ cdef class DatasetsPair{{name_suffix}}:
         # TODO: add "with gil: raise" here when supporting Cython 3.0
         return -1
 
+
+@final
+cdef class PrecomputedDistanceMatrix{{name_suffix}}(DatasetsPair{{name_suffix}}):
+    """A precomputed distance matrix between row vectors of two arrays.
+    Its abstract class is DatasetsPair
+
+    Parameters
+    ----------
+    Must receive precomputed_distance: ndarray of shape (n_samples_X, n_samples_Y)? Really? 
+        Must be C-contiguous. 
```

Comment:
> Really?

IMO, it would make sens to support C-contiguity for now. We can always extend it then.